src/data/nat/basic.lean

```diff
@@ -118,6 +118,40 @@ end facts
 namespace nat
 variables {m n k : ℕ}
 
+/-!
+### Recursion and `set.range`
+-/
+
+section set
+
+open set
+
+theorem zero_union_range_succ : {0} ∪ range succ = univ :=
+by { ext n, cases n; simp }
+
+variables {α : Type*}
+
+theorem range_of_succ (f : ℕ → α) : {f 0} ∪ range (f ∘ succ) = range f :=
+by rw [← image_singleton, range_comp, ← image_union, zero_union_range_succ, image_univ]
+
+theorem range_rec {α : Type*} (x : α) (f : ℕ → α → α) :
+  set.range (show ℕ → α, from λ n, nat.rec x f n) =
```

Comment:
What about `(set.range (λ n, nat.rec x f n) : set α)` instead? It looks more readable to me. Same thing in the next lemma. Choose whichever you prefer, and then you can merge.